toxcore/util.c

```diff
@@ -80,6 +81,21 @@ bool memeq(const uint8_t *a, size_t a_size, const uint8_t *b, size_t b_size)
     return a_size == b_size && memcmp(a, b, a_size) == 0;
 }
 
+uint8_t *memdup(const uint8_t *data, size_t data_size)
+{
+    if (data_size == 0) {
```

Comment:
This should probably also check if `data == nullptr`?

Reply:
Yes good call. Done.